requests/models.py

```diff
@@ -420,8 +420,11 @@ def prepare_body(self, data, files, json=None):
         length = None
 
         if not data and json is not None:
-            content_type = 'application/json'
-            body = complexjson.dumps(json)
+            # When urllib3 uses pyOpenSSL, it can only resume large uploads
+            # properly if receiving a bytes-like object. In Python 2, json.dumps()
+            # returns just that, but Python 3 returns a Unicode string.
+            content_type = 'application/json; charset=utf-8'
```

Comment:
Fascinating!

>3.  Encoding

>   JSON text SHALL be encoded in Unicode.  The default encoding is UTF-8.

>   Since the first two characters of a JSON text will always be ASCII
   characters [RFC0020], it is possible to determine whether an octet
   stream is UTF-8, UTF-16 (BE or LE), or UTF-32 (BE or LE) by looking
   at the pattern of nulls in the first four octets.

 >            00 00 00 xx  UTF-32BE
           00 xx 00 xx  UTF-16BE
           xx 00 00 00  UTF-32LE
           xx 00 xx 00  UTF-16LE
           xx xx xx xx  UTF-8

Fix incoming.